src/project/manifest/mod.rs

```diff
@@ -742,6 +742,47 @@ impl ProjectManifest {
     }
 }
 
+#[derive(Debug, Clone, Eq)]
+pub(crate) struct UniquePackageName {
+    inner: PackageName,
+}
+
+impl PartialEq for UniquePackageName {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_inner().eq(&other.as_inner())
+    }
+}
+
+impl Hash for UniquePackageName {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.as_inner().as_normalized().hash(state);
+    }
+}
+
+impl UniquePackageName {
+    pub fn as_inner(&self) -> PackageName {
+        self.inner.clone()
+    }
+}
+
+impl<'de> Deserialize<'de> for UniquePackageName {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let package_name = PackageName::deserialize(deserializer)?;
+        if package_name.as_source() != package_name.as_normalized() {
+            Err(serde::de::Error::custom(
+                "invalid dependency: please avoid using capitalized names for the dependencies",
```

Comment:
```suggestion
                "invalid dependency: cannot use capitalized names for the dependencies",
```
Small change, `please` didn't sound like a error, this was my fault!